lib/libzfs/libzfs_dataset.c

```diff
@@ -2979,6 +2980,11 @@ zfs_prop_get_int(zfs_handle_t *zhp, zfs_prop_t prop)
 	char *source;
 	uint64_t val = 0;
 
+	if ((prop == ZFS_PROP_CREATETXG) &&
+	    ((val = zfs_get_createtxg(zhp)) > 0)) {
+		return (val);
+	}
+
```

Comment:
Avoiding double conversion is good, but it is not added into get_numeric_property()?  I see few other properties handled there the same way.  Turning this function from trivial wrapper into something separate looks wrong to me.

Reply:
Totally agreed. Updated!